src/analysis/inner_product_space/linear_pmap.lean

```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2022 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll
+-/
+
+import analysis.inner_product_space.basic
+import topology.algebra.module.linear_pmap
+import analysis.inner_product_space.dual
+
+/-!
+# Partially defined linear operators on Hilbert spaces
+
+We will develop the basics of the theory of unbounded operators on Hilbert spaces.
+
+## Main definitions
+
+* `linear_pmap.is_formal_adjoint`: An operator `T` is a formal adjoint of `S` if for all `x` in the
+domain of `T` and `y` in the domain of `S`, we have that `⟪T x, y⟫ = ⟪x, S y⟫`.
+* `linear_pmap.adjoint`: The adjoint of a map `E →ₗ.[𝕜] F` as a map `F →ₗ.[𝕜] E`.
```

Comment:
In the definition below, don't you assume that the initial map is densely defined?

By the way, I am not completely sure about this design: can't you define a meaningful adjoint when the map is not densely defined, as follows: let A be the closure of its domain. Then you have a densely defined map from A to F, and therefore an adjoint from F to A, that you can extend by composition to an adjoint from F to E. This looks like a canonical enough definition even when the map is not densely defined, no? (And probably you can give a direct definition for it, instead of going through the factorization I have just described).

Reply:
A similar way of seeing this construction could be: you can extend any non-densely defined operator T to a densely defined one by decomposing the space into the direct sum of A and the orthogonal complement of A and extending T to 0 on the orthogonal component and then using the construction of this PR on there. I would be curious to know a situation where the adjoint of a non-densely defined operator is needed though, I have never seen it before!

Reply:
The problem is that we need that the space `A` has to be complete to apply Fréchet-Riesz. In fact there is a different way to construct the adjoint using its graph and the 'graph' gives rise to an operator iff the domain of `T` is dense.